spec/pageflow/policies/folder_policy_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'spec_helper'
+
+module Pageflow
+  module Policies
+    describe FolderPolicy do
+      terms = {class: FolderPolicy,
+               subject: FactoryGirl.create(:folder),
+               name: 'folder'}
+      terms = terms.merge!(entity: terms[:subject].account)
+      describe :manage do
+        it_behaves_like 'a membership-based permission that',
+                        terms: terms,
+                        entity_type: :account,
+                        permission_type: :manage,
+                        minimum_required_role: 'publisher',
+                        maximum_forbidden_role: 'editor'
+      end
+    end
```

Comment:
I still find this a bit hard to read. Especially the `terms` hash makes it more opaque. According to [RSpec docs](https://www.relishapp.com/rspec/rspec-core/v/2-0/docs/example-groups/shared-example-group#shared-example-group-applied-to-two-groups) the `described_class` is visible inside shared examples. So no  need to pass the policy. What do you think of making the parameters read more like a sentence. That would also work nicely with the trailing "that" in the description:

```
describe FolderPolicy do
   it_behaves_like 'a membership-based permission that',
                            allows: 'publisher',
                            but_forbids: 'editor',
                            of: -> { |subject| subject.account }
                            to: :manage,
                            subject: -> { create(:folder) }
end
```

That way we make the policy subject explicit. Moreover, we make explicit how the membership entity shall be obtained from the subject. `name` parameter can be obtained from `params[:subject].class.name`. The entity name is `params[:of].class.name`